### DISCUSSION TITLE
Change data types by C# script

### DISCUSSION THREAD

**Discussion by rvgfox** [TabularEditor/TabularEditor3#1163] [🙋♀️TE3 Q&A] [answered]
I'm creating the models in an Azure Analysis Server using an "automatic" tool. This tool create all the numeric colums using the Double data type.
I've a simple C# script to change the selected columns data type from Double to Decimal:

```
// Loop through the list of selected columns
foreach(var c in Selected.Columns)
{
 
    // Hide the base column:
    c.IsHidden = true;
    
    //Change date type
    c.DataType=DataType.Decimal;
}
```

Two questions:

1. Do you think that it's necessary to reduce the model size?
2. How can I change the C# script for change the data type of all columns not the selected ones?

Thanks in advance

**Comment by otykier** [comment, marked as answer]
1. This change is not guaranteed to reduce the model size. That depends on the cardinality (i.e. number of unique values) within the column. If you have a lot of values that differ only by very small decimal digits (i.e. less than the first 4 decimals), then this change can reduce the size of the column in memory, at the cost of less precision. In general, floating point values (double) is not recommended unless you have a specific need for it. [More info here](https://www.sqlbi.com/articles/choosing-numeric-data-types-in-dax/).
2. Use `foreach(var c in Model.AllColumns)`. However, you should probably add a check to make sure you're only changing columns that are currently set to Double:

```csharp
// Loop through all columns of the model:
foreach (var c in Model.AllColumns)
{
    // If a column is currently set to double...
    if (c.DataType == DataType.Double)
    {
        // ...hide the column:
        c.IsHidden = true;
    
        // ...change date type to decimal:
        c.DataType = DataType.Decimal;
    }
}
```

**Reply by rvgfox** [reply]
Thanks Daniel.

I know that the use of Double data type it's not recomended, but the problem it's that the tool that I'm using to build the model use it.

The script that you share with me (thanks a lot of) solve my problem because all the numbers that I'm using are money and quantities and two decimals are enought.
